test/regress_listener.c

```diff
@@ -304,6 +309,72 @@ regress_listener_error_unlock(void *arg)
 }
 #endif
 
+#ifndef EVENT__DISABLE_THREAD_SUPPORT
+
+static THREAD_FN
+disable_thread(void * arg)
+{
+    struct evconnlistener *lev = (struct evconnlistener *)arg;
+    evconnlistener_disable(lev);
+    return NULL;
+}
+
+static void
+acceptcb_for_thread_test(struct evconnlistener *listener, evutil_socket_t fd,
+    struct sockaddr *addr, int socklen, void *arg)
+{
+	THREAD_T *threadid_arg;
+	THREAD_T threadid;
+
+	threadid_arg = (THREAD_T *)arg;
+
+	evutil_closesocket(fd);
+
+	THREAD_START(threadid, disable_thread, listener);
+	sleep(1);
```

Comment:
Relying on test --timeout is not great, let's use your initial reproducer, but decrease delay 10x

Reply:
Actually I've no idea about how to reproduce this issue more gently.  I have tried to add a notify event to tell main thread that the `disable_thread` is already startingup and going to call `evconnlistener_disable()` but it does not ensure the locking action. So I have to add a sleep here. But change `sleep` to `evutil_usleep_`  can siply decrese.  I'm looking forward to hearing from you for any suggestion.

Reply:
>But change sleep to evutil_usleep_ can siply decrese. I'm looking forward to hearing from you for any suggestion.

Yes, let's simply decrease the delay, this will be enough for now.

AFAICS the problem here is that then disable_thread is executed, the event loop cannot be broken, due to the lock is held, so you cannot notify in any other form (let's also put a comment about this in the code).